### DISCUSSION TITLE
Refine tetrahedral elements

### DISCUSSION THREAD

**Discussion by VictorS-M** [simnibs/simnibs#395] [7.Creating head models] [answered]
Hello everyone.
I am trying to visualise the electric field in the grey matter volume and I notice that the tetrahedral elements in the cerebellum are not smooth as I wanted (I attach a picture).
![Captura de pantalla 2024-07-22 112425](https://github.com/user-attachments/assets/0c0aceda-b3dc-4182-9838-d7a1bd6e275c)

I have two questions about this:
- Does this affect the calculation of the electric field?
- How can I refine this part of the grey matter?

Thank you very much.

**Comment by mabe2416** [comment]
If the mesh elements are generally too large you can adapt the settings file charm.ini for the tissue you want to refine and then run the meshing again. 

But as I understood, you only want to refine the mesh in a certain ROI. 

The option I use for this is the pynibs package. 
You can have a look at the function pynibs.create_refine_spherical_roi(), where you input the final_tissues.nii.gz of your mesh and give the centre coordinate and radius of a spherical roi, where you want to refine the mesh. The output sizefield can then be used as an input for --sizing_field in SimNIBS-4/simnibs/cli/meshmesh.py. 

I don't know if there is a simpler approach, since Simnibs 4 can already output sizefield.nii.gz. Maybe someone knows a simple way to modify this sizefield to give it as input to meshmesh again?

**Comment by bjornsig** [comment, marked as answer]
Hi,

Please ensure that the saturate setting is enabled otherwise elements above the color range will not be shown which might be the reason for the course looking surface in this case.

![image](https://github.com/user-attachments/assets/ba4f3a8d-d75f-42a1-89bf-cac528d1ae95)

**Comment by VictorS-M** [comment]
Thank you for your answers.
I just realised it was a display problem.
